### PR TITLE
Rya 104 mongo admin shell

### PR DESCRIPTION
**Pull request by isper3at**
## Description

Add mongo support to admin shell
### Links

[Jira](https://issues.apache.org/jira/browse/RYA-104)
### Checklist
- [ ] Code Review
- [ ] Squash commits
#### People To Review

@kchilton2

### CODE REVIEW

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java [hunk 1]
```diff
@@ -35,13 +35,17 @@
...
 import com.mongodb.MongoClient;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
 
 import mvm.rya.shell.SharedShellState.ConnectionState;
 import mvm.rya.shell.command.CommandException;
 import mvm.rya.shell.command.RyaCommands;
 import mvm.rya.shell.command.accumulo.AccumuloConnectionDetails;
 import mvm.rya.shell.command.administrative.InstanceExists;
+import mvm.rya.shell.command.mongo.MongoConnectionDetails;
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: You never updated the public String printConnectionDetails() method in this class to print Mongo specific details if that is what you're connected to.

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java [hunk 2]
```diff
@@ -154,24 +158,27 @@ public String connectToAccumulo(
 
     @CliCommand(value = CONNECT_MONGO_CMD, help = "Connect the shell to an instance of Mongo.")
     public String connectToMongo(
-            @CliOption(key = {"username"}, mandatory = true, help = "The username that will be used to connect to Accummulo.")
+            @CliOption(key = {"username"}, mandatory = true, help = "The username that will be used to connect to Mongo.")
             final String username,
-            @CliOption(key = {"instanceName"}, mandatory = true, help = "The name of the Accumulo instance that will be connected to.")
+            @CliOption(key = {"hostname"}, mandatory = true, help = "The hostname where Mongo resides.")
+            final String hostname,
+            @CliOption(key = {"instanceName"}, mandatory = true, help = "The name of the Mongo instance that will be connected to.")
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: What is a Mongo instance name? We only have this for Accumulo because the Accumulo instance itself has a name within zookeeper.

Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: According to Mongo DB documentation, that's the database name. I think the Database Name is the same as the Rya Instance Name, which is not what you should be connecting to with this command. This should indicate an instance of Mongo to connect to so that you can list the rya instances that are hosted on it.

Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: public void connectToInstance(...) is where you specify which rya instance will be interacted with.

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java [hunk 3]
```diff
@@ -154,24 +158,27 @@ public String connectToAccumulo(
...
-            final MongoClient client = new Mongo
+            final MongoCredential creds = MongoCredential.createCredential(username, instanceName, password);
+            final MongoClient client = new MongoClient(new ServerAddress(hostname), Lists.newArrayList(creds));
 
-            // Initialize the connected to Accumulo shared state.
-            final AccumuloConnectionDetails accumuloDetails = new AccumuloConnectionDetails(username, password, instanceName, zookeepers, auths.toString());
-            final RyaCommands commands = RyaCommands.buildAccumuloCommands(accumuloDetails, connector, auths);
-            sharedState.connectedToAccumulo(accumuloDetails, commands);
+            // Initialize the connected to Mongo shared state.
+            final MongoConnectionDetails accumuloDetails = new MongoConnectionDetails(username, password, instanceName, hostname);
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: Bad variable name.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/MongoConnectionDetails.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    private final char[] password;
+    private final String instanceName;
+    private final String hostname;
+
+    /**
+     * Constructs an instance of {@link MongoConnectionDetails}.
+     *
+     * @param username - The username that was used to establish the connection. (not null)
+     * @param password - The password that was used to establish the connection. (not null)
+     * @param instanceName - The Mongo instance name that was used to establish the connection. (not null)
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/MongoConnectionDetails.java]: I don't think this is a thing.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/MongoCommand.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+     *   that hosts Rya instance. (not null)
+     */
+    public MongoCommand(
+            final MongoConnectionDetails connectionDetails,
+            final MongoClient client) {
+        this.connectionDetails = requireNonNull( connectionDetails );
+        this.client = requireNonNull(client);
+    }
+
+    public MongoConnectionDetails getMongoConnectionDetails() {
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/MongoCommand.java:54]: Docs.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoCreatePCJ.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import mvm.rya.shell.command.mongo.MongoCommand;
+import mvm.rya.shell.command.mongo.MongoConnectionDetails;
+
+// TODO Impl, test
+
+/**
+ * An Mongo implementation of the {@link CreatePCJ} command.
+ */
+@ParametersAreNonnullByDefault
+public class MongoCreatePCJ extends MongoCommand implements CreatePCJ {
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoCreatePCJ.java:35]: Document why this command isn't supported for Mongo.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoCreatePCJ.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+     *   that hosts Rya instance. (not null)
+     */
+    public MongoCreatePCJ(final MongoConnectionDetails connectionDetails, final MongoClient client) {
+        super(connectionDetails, client);
+    }
+
+    @Override
+    public String createPCJ(final String instanceName, final String sparql) throws InstanceDoesNotExistException, CommandException {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Not implemented yet.");
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoCreatePCJ.java]: Update this text and remove TODOs.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoDeletePCJ.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+import mvm.rya.shell.command.mongo.MongoCommand;
+import mvm.rya.shell.command.mongo.MongoConnectionDetails;
+
+// TODO impl, test
+
+/**
+ * An Mongo implementation of the {@link DeletePCJ} command.
+ */
+@ParametersAreNonnullByDefault
+public class MongoDeletePCJ extends MongoCommand implements DeletePCJ {
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoDeletePCJ.java:35]: Same comment as add pcj,

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java (hunk 1 above)
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: On that note, probs need integration tests to make sure the console works with mongo.

Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: https://github.com/kchilton2/incubator-rya/blob/RYA-82/extras/rya.console/src/test/java/mvm/rya/shell/RyaConnectionCommandsIT.java

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/MongoConnectionDetails.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * The information that the shell used to connect to Mongo.
+ */
+@Immutable
+@ParametersAreNonnullByDefault
+public class MongoConnectionDetails {
+    private final String username;
+    private final char[] password;
+    private final String instanceName;
+    private final String hostname;
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/MongoConnectionDetails.java]: You can only host multiple instances of Mongo is if they're running on different ports. If we want to support multiple mongo instances (which I don't think we do right now), include a port here instead of instanceName.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoListInstances.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+     *   that hosts Rya instance. (not null)
+     */
+    public MongoListInstances(final MongoConnectionDetails connectionDetails, final MongoClient client) {
+        super(connectionDetails, client);
+    }
+
+    @Override
+    public List<String> listInstances() throws CommandException {
+        //TODO: Exclude system dbs.
+        return super.getClient().getDatabaseNames();
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoListInstances.java]: You already have a TODO here, but yea, this will not work. You need to make sure each of these DBs are actually Rya DBs. What if your mongo instance also hosts a bunch of other applications?

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java (hunk 1 above)
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: Also, RyaAdminCommands is going to need to be updated to only show the supported commands. Since Mongo DB will not support add/remove PCJ, then they shouldn't be presented as options to the user.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+        conf.set(RdfCloudTripleStoreConfiguration.CONF_TBL_PREFIX, details.getRyaInstanceName());
+
+        // Enable the indexers that the instance is configured to use.
+        conf.set(ConfigUtils.USE_PCJ, "" + details.getPCJIndexDetails().isEnabled() );
+        conf.set(ConfigUtils.USE_GEO, "" + details.getGeoIndexDetails().isEnabled() );
+        conf.set(ConfigUtils.USE_FREETEXT, "" + details.getFreeTextIndexDetails().isEnabled() );
+        conf.set(ConfigUtils.USE_TEMPORAL, "" + details.getTemporalIndexDetails().isEnabled() );
+        conf.set(ConfigUtils.USE_ENTITY, "" + details.getEntityCentricIndexDetails().isEnabled());
+
+        // XXX The Mongo implementation of the secondary indices make need all
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java]: Do you actually need this on the Mongo side? This looks like copy paste from accumulo.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+     *   operation to fail.
+     */
+    private RyaDetails initializeRyaDetails(final String instanceName, final InstallConfiguration installConfig)
+            throws AlreadyInitializedException, RyaDetailsRepositoryException {
+        final RyaDetailsRepository detailsRepo = new MongoRyaInstanceDetailsRepository(getClient(), instanceName);
+
+        // Build the PCJ Index details.
+        final PCJIndexDetails.Builder pcjDetailsBuilder = PCJIndexDetails.builder()
+                .setEnabled(installConfig.isPcjIndexEnabled());
+        if(installConfig.getFluoPcjAppName().isPresent()) {
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java]: Is it legitimate to include a fluo app name when you're using mongo? Maybe the install prompt should be different for mongo (although, there nothing stopping you from using fluo to update a mongo instance)

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+import mvm.rya.shell.command.mongo.MongoCommand;
+import mvm.rya.shell.command.mongo.MongoConnectionDetails;
+
+// TODO impl, test
+
+/**
+ * An Mongo implementation of the {@link Uninstall} command.
+ */
+@ParametersAreNonnullByDefault
+public class MongoUninstall extends MongoCommand implements Uninstall {
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java:36]: This seems easy to implement on the Mongo side since you just drop a DB, so I think we should support it.

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java [hunk 14]
```diff
@@ -16,6 +16,24 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+/**
```
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java]: looks like another license was added

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstanceExists.java [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+     *   that hosts Rya instance. (not null)
+     */
+    public MongoInstanceExists(final MongoConnectionDetails connectionDetails, final MongoClient client) {
+        super(connectionDetails, client);
+    }
+
+    @Override
+    public boolean exists(final String instanceName) throws CommandException {
+        requireNonNull( instanceName );
+        final List<String> dbNames = super.getClient().getDatabaseNames();
```
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstanceExists.java]: remove this line

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoListInstances.java [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+        super(connectionDetails, client);
+    }
+
+    @Override
+    public List<String> listInstances() throws CommandException {
+        final MongoClient client = super.getClient();
+        final List<String> dbNames = client.getDatabaseNames();
+        final List<String> ryaInstances = new ArrayList<>();
+        for(final String db : dbNames) {
+            final Set<String> collNames = client.getDB(db).getCollectionNames();
```
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoListInstances.java:57]: or check for details

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java [hunk 17]
```diff
@@ -125,8 +126,10 @@ public boolean arePCJCommandsAvailable() {
         // that is new enough to use the RyaDetailsRepository and is configured to maintain PCJs.
         final ShellState shellState = state.getShellState();
         if(shellState.getConnectionState() == ConnectionState.CONNECTED_TO_INSTANCE) {
-            //TODO: if Mongo, return false.
             final GetInstanceDetails getInstanceDetails = shellState.getConnectedCommands().get().getGetInstanceDetails();
+            if(getInstanceDetails instanceof MongoGetInstanceDetails) {
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java]: This is kind of a hack. Why not just store the type of repository we're connected to in the shell state object and key off of that?

File: dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java [hunk 18]
```diff
@@ -167,7 +164,7 @@ public static RyaDetails toRyaDetails(final DBObject mongoObj) throws MalformedR
         return RyaDetails.builder()
             .setRyaInstanceName(basicObj.getString(INSTANCE_KEY))
             .setRyaVersion(basicObj.getString(VERSION_KEY))
-            .setEntityCentricIndexDetails(new EntityCentricIndexDetails(basicObj.getBoolean(ENTITY_DETAILS_KEY)))
+            .setEntityCentricIndexDetails(new EntityCentricIndexDetails(false))//entity indexing is not supported
             .setGeoIndexDetails(new GeoIndexDetails(basicObj.getBoolean(GEO_DETAILS_KEY)))
             .setPCJIndexDetails(getPCJIndexDetails(basicObj))
```
Comment (kchilton2) [inline dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java:172]: PCJs also aren't supported by the Mongo version of this stuff, so it's kind of weird that you've forced the entity centric stuff to always be false, but actually deserialize a DBObject here for the PCJ stuff. I think it makes more sense to handle that nuance outside of this class. Either always make sure these field are disabled before you even serialize them or change the RyaDetails to make these indices optional.

I dunno. This is probably worth having a conversation about. Mongo and Accumulo having different features isn't something we planned for.

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java (hunk 17 above)
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java]: didn't know if you wanted me messing with ShellState

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java [hunk 19]
```diff
@@ -146,7 +146,7 @@ private RyaDetails initializeRyaDetails(final String instanceName, final Install
             .setGeoIndexDetails(new GeoIndexDetails(installConfig.isGeoIndexEnabled()))
             .setTemporalIndexDetails(new TemporalIndexDetails(installConfig.isTemporalIndexEnabled()))
             .setFreeTextDetails(new FreeTextIndexDetails(installConfig.isFreeTextIndexEnabled()))
-            .setEntityCentricIndexDetails(new EntityCentricIndexDetails(installConfig.isEntityCentrixIndexEnabled()))
+            .setEntityCentricIndexDetails(new EntityCentricIndexDetails(false))//not supported in mongo
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java:149]: Did you update the shell's Install prompt so that it won't even offer EntityCentric and PCJ indices as options when you're connected to Mongo?

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java [hunk 20]
```diff
@@ -108,6 +127,9 @@ public boolean arePCJCommandsAvailable() {
         final ShellState shellState = state.getShellState();
         if(shellState.getConnectionState() == ConnectionState.CONNECTED_TO_INSTANCE) {
             final GetInstanceDetails getInstanceDetails = shellState.getConnectedCommands().get().getGetInstanceDetails();
+            if(getInstanceDetails instanceof MongoGetInstanceDetails) {
+                return false;
+            }
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java]: This is kind of a hack. Store what type of Rya storage we are connected to within the shared shell state and then key off of that.

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java [hunk 21]
```diff
@@ -109,6 +106,7 @@ public boolean isDisconnectAvailable() {
     public String printConnectionDetails() {
         final Optional<AccumuloConnectionDetails> detailsHolder = sharedState.getShellState().getConnectionDetails();
 
+        //TODO: MONGO
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: Looks like you haven't addressed this TODO yet.

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java [hunk 22]
```diff
@@ -164,8 +162,8 @@ public String connectToMongo(
             final MongoClient client = new MongoClient(new ServerAddress(hostname), Lists.newArrayList(creds));
 
             // Initialize the connected to Mongo shared state.
-            final MongoConnectionDetails accumuloDetails = new MongoConnectionDetails(username, password, instanceName, hostname);
-            final RyaCommands commands = RyaCommands.buildMongoCommands(accumuloDetails, client);
+            final MongoConnectionDetails mongoDetails = new MongoConnectionDetails(username, password, instanceName, hostname);
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: Why do we need instance name here again?

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java [hunk 23]
```diff
@@ -134,36 +133,26 @@ private RyaDetails initializeRyaDetails(final String instanceName, final Install
             throws AlreadyInitializedException, RyaDetailsRepositoryException {
         final RyaDetailsRepository detailsRepo = new MongoRyaInstanceDetailsRepository(getClient(), instanceName);
 
-        // Build the PCJ Index details.
+        // Build the PCJ Index details. [not supported in mongo]
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java:136]: I think we should throw an exception if the user is trying to install an instance with PCJ turned on while it's not supported.

The install prompt shouldn't even provide the option to the user.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java [hunk 24]
```diff
@@ -134,36 +133,26 @@ private RyaDetails initializeRyaDetails(final String instanceName, final Install
...
-                .build();
+            // General Metadata
+            .setRyaInstanceName(instanceName)
+            .setRyaVersion( getVersion() )
+
+            // Secondary Index Values
+            .setGeoIndexDetails(new GeoIndexDetails(installConfig.isGeoIndexEnabled()))
+            .setTemporalIndexDetails(new TemporalIndexDetails(installConfig.isTemporalIndexEnabled()))
+            .setFreeTextDetails(new FreeTextIndexDetails(installConfig.isFreeTextIndexEnabled()))
+            .setEntityCentricIndexDetails(new EntityCentricIndexDetails(false))//not supported in mongo
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java:149]: Throw exception.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstanceExists.java [hunk 25]
```diff
@@ -49,6 +51,7 @@ public MongoInstanceExists(final MongoConnectionDetails connectionDetails, final
     @Override
     public boolean exists(final String instanceName) throws CommandException {
         requireNonNull( instanceName );
+        final List<String> dbNames = super.getClient().getDatabaseNames();
         return super.getClient().getDatabaseNames().contains(instanceName);
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstanceExists.java:52]: Duplicate code.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoListInstances.java (hunk 16 above)
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoListInstances.java:57]: I think we should just trigger off of details. I think puja dropped the reverse compatibility requirement for the shell.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java [hunk 26]
```diff
@@ -48,9 +47,12 @@ public MongoUninstall(final MongoConnectionDetails connectionDetails, final Mong
...
     @Override
-    public void uninstall(final String instanceName) throws InstanceDoesNotExistException, CommandException {
-        // TODO Auto-generated method stub
-
-        throw new UnsupportedOperationException("Not implemented yet.");
+    public void uninstall(final String ryaInstanceName) throws InstanceDoesNotExistException, CommandException {
+        try {
+            // TODO impl, test
+            getClient().dropDatabase(ryaInstanceName);
+        }catch (final MongoException e) {
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java:56]: Are you able to tell if this should be an InstanceDoesNotExistException?

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java [hunk 27]
```diff
@@ -48,9 +47,12 @@ public MongoUninstall(final MongoConnectionDetails connectionDetails, final Mong
     }
 
     @Override
-    public void uninstall(final String instanceName) throws InstanceDoesNotExistException, CommandException {
-        // TODO Auto-generated method stub
-
-        throw new UnsupportedOperationException("Not implemented yet.");
+    public void uninstall(final String ryaInstanceName) throws InstanceDoesNotExistException, CommandException {
+        try {
+            // TODO impl, test
```
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java]: Address test TODO or remove comment if it is done.

File: extras/rya.console/src/test/java/mvm/rya/shell/MongoITBase.java [hunk 28]
```diff
@@ -0,0 +1,85 @@
...
+import de.flapdoodle.embed.mongo.tests.MongodForTestsFactory;
+import mvm.rya.shell.command.mongo.MongoConnectionDetails;
+
+/**
+ * The base class for any integration test that needs to use a {@link MiniAccumuloCluster}.
+ */
+public class MongoITBase {
+
+    /**
+     * TODO doc
```
Comment (kchilton2) [inline extras/rya.console/src/test/java/mvm/rya/shell/MongoITBase.java]: TODO

File: extras/rya.console/src/test/java/mvm/rya/shell/MongoITBase.java [hunk 29]
```diff
@@ -0,0 +1,85 @@
...
+    private static MongoConnectionDetails connectionDetails = null;
+
+    @BeforeClass
+    public static void startMongo() throws IOException, InterruptedException, MongoException {
+        final MongodForTestsFactory factory = new MongodForTestsFactory();
+        client = factory.newMongo();
+        connectionDetails = new MongoConnectionDetails(
+            "username",
+            "pswd".toCharArray(),
+            "rya",
```
Comment (kchilton2) [inline extras/rya.console/src/test/java/mvm/rya/shell/MongoITBase.java:52]: Why do we need the "rya" string here?

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java (hunk 20 above)
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaAdminCommands.java]: Also, did you write/update any mongo shell tests?

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java (hunk 22 above)
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: because you can't connect to mongo without knowing which database to use.  the databases are named with instance names

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoListInstances.java (hunk 16 above)
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoListInstances.java:57]: only triggers off of details now.

File: dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java (hunk 18 above)
Comment (isper3at) [inline dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java:172]: I did it this way as a catch all placeholder for now.  it was the fastest way to get the job done.  A conversation does need having

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstanceExists.java (hunk 25 above)
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstanceExists.java:52]: yeah I had commented further up to remove that line, forgot to. merp.

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java (hunk 26 above)
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java:56]: no, the only way to do that would be to get the list of database names and check for exists.  if not, throw expection.  I can do that if you want me to

File: extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java (hunk 27 above)
Comment (isper3at) [inline extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoUninstall.java]: done

File: extras/rya.console/src/test/java/mvm/rya/shell/MongoITBase.java (hunk 28 above)
Comment (isper3at) [inline extras/rya.console/src/test/java/mvm/rya/shell/MongoITBase.java]: just removed, don't see why it needs docs.  copy paste artifact from your AccumuloITBase

File: extras/rya.console/src/test/java/mvm/rya/shell/MongoITBase.java (hunk 29 above)
Comment (isper3at) [inline extras/rya.console/src/test/java/mvm/rya/shell/MongoITBase.java:52]: its the test rya instance name to use

File: extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java (hunk 22 above)
Comment (kchilton2) [inline extras/rya.console/src/main/java/mvm/rya/shell/RyaConnectionCommands.java]: If this is also going to connect to an instance of Rya, then you need to update the shared shell state to indicate it's connected to an instance. However, this deviates from how the Accumulo side works, so this may cause user's confusion.

File: dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java (hunk 18 above)
Comment (kchilton2) [inline dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java:172]: Since you removed the integration code, could we just assume it should be read from the db object here? This is going to be really confusing code otherwise.

File: dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java [hunk 30]
```diff
@@ -180,20 +177,16 @@ public static RyaDetails toRyaDetails(final DBObject mongoObj) throws MalformedR
...
     }
 
+    /**
+     * @param basicObj
+     * @return A builder setting the PCJIndexer to false since Mongo does not support PCJ indexing.
+     */
     private static PCJIndexDetails.Builder getPCJIndexDetails(final BasicDBObject basicObj) {
         final BasicDBObject pcjIndexDBO = (BasicDBObject) basicObj.get(PCJ_DETAILS_KEY);
 
         final PCJIndexDetails.Builder pcjBuilder = PCJIndexDetails.builder()
```
Comment (kchilton2) [inline dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java:186]: Revert this change too. Even though we don't currently support PCJs, I think that nuance should be encapsulated elsewhere.

Comment (isper3at) [inline dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java:186]: I reverted the whole file since it seemed all of the changes were like this
